docs/faq/add_op_in_backend.md

```diff
@@ -543,53 +545,81 @@ If you use python, when you type `import mxnet as mx`, two python
+the other one is for symbolic programming registered under
+module `mxnet.symbol.quadratic` or `mx.sym.quadratic` for short.
 
 In order to unit test it in frontend, we need to add the following code
-to the python file `test_operator.py`. Note that while testing the
-forward pass is straightforward using `mx.nd.quadratic`, testing
-the backward involves a bit of more efforts. We create a
-`quadratic` symbol and feed it into the utility function `check_numeric_gradient`.
-The utility function will perform a perturbation on the input
-and calculate the response rate of the output using the
-[finite difference method](https://en.wikipedia.org/wiki/Finite_difference_method).
-Then it will compare the gradient from the backward pass with
-the values from the finite difference method. The test
-will be successful once the comparison satisfies user specified
-relative and absolute thresholds.
+to the python file `test_operator.py`. A typical operator implementation
+tests for both the `symbol` API and the `ndarray` API. In this tutorial 
+we will go over the symbol API test. Note that while testing
+the `ndarray` API is pretty straight forward as we are testing only the
+forward pass of the operator, testing for the `symbol` API is more involved
+as it tests the complete functionality of the operator - the forward pass,
+the backward pass and the gradient computation in the operator implementation.
+To facilitate the testing of these functionalities we use three helper
+functions available in the `mxnet.test_utils` module:
+ - `check_symbolic_forward `
+ - `check_symbolic_backward`
+ - `check_numeric_gradient`
+
 ```python
 def test_quadratic_function():
     def f(x, a, b, c):
         return a * x**2 + b * x + c
 
+    ctx = default_context()
     a = np.random.random_sample()
     b = np.random.random_sample()
     c = np.random.random_sample()
-    for ndim in range(1, 6):
-        # check forward
-        shape = rand_shape_nd(ndim, 5)
-        data = rand_ndarray(shape=shape, stype='default')
-        data_np = data.asnumpy()
-        expected = f(data_np, a, b, c)
-        output = mx.nd.quadratic(data, a=a, b=b, c=c)
-        assert_almost_equal(output.asnumpy(), expected)
-
-        # check backward using finite difference
-        data = mx.sym.Variable('data')
-        quad_sym = mx.sym.quadratic(data=data, a=a, b=b, c=c)
-        check_numeric_gradient(quad_sym, [data_np])
+    data = mx.symbol.Variable('data')
+    quad_sym = mx.sym.contrib.quadratic(data=data, a=a, b=b, c=c)
+    for dtype in [np.float16, np.float32, np.float64]:
+        for ndim in range(1, 6):
+            shape = rand_shape_nd(ndim, 5)
+            data_np = np.random.randn(*shape).astype(dtype)
+            expected = f(data_np, a, b, c)
+            backward_expected = 2 * a * data_np + b
+            # check forward
+            check_symbolic_forward(quad_sym, [data_np], [expected],
+                                    rtol=1e-2 if dtype is np.float16 else 1e-5,
```

Comment:
Have you actually run through this example and verified that this operator indeed needs such tolerance levels?

Reply:
yes.